lib/ddsketch-agent/src/lib.rs

```diff
@@ -434,16 +434,31 @@ impl DDSketch {
 
         let key = self.config.key(v);
 
-        // Fast path for adding to an existing bin.
-        for b in &mut self.bins {
-            if b.k == key && b.n < MAX_BIN_WIDTH {
-                b.n += 1;
-                return;
+        let mut insert_at = None;
+
+        for (bin_idx, b) in self.bins.iter_mut().enumerate() {
+            if b.k == key {
+                if b.n < MAX_BIN_WIDTH {
+                    // Fast path for adding to an existing bin without overflow.
+                    b.n += 1;
+                    return;
+                } else {
+                    insert_at = Some(bin_idx);
```

Comment:
This might be entirely irrelevant, but do you think it makes any difference if we insert the new bin at the existing index, or after the existing index?

My mind naturally wants to see 0:65535 0:1, ordered largest to smallest, but then I wondered if it actually may matter to downstream systems or not. Guessing not, but it feels like you're very up-to-speed here and might have a good intuition of if it does or doesn't matter.

Reply:
I'm not aware of any downstream requirement on the bin order, but we do depend on the incremental counts coming first in `insert_keys` and `trim_left`.

Reply:
Ah, good point. 👍🏻